src/rule_tester.rs

```diff
@@ -31,55 +37,55 @@ impl RuleTester {
     }
 
     fn run_valid_test(&self, valid_test: &RuleTestValid) {
-        let tmp_dir = TempDir::new("valid_test").unwrap();
-        let test_filename = "tmp.rs";
-        fs::write(tmp_dir.path().join(test_filename), &valid_test.code).unwrap();
-        Command::cargo_bin("tree-sitter-lint")
-            .unwrap()
-            .args(["--rule", &self.rule.name])
-            .current_dir(tmp_dir.path())
-            .assert()
-            .success()
-            .stdout(predicate::str::is_empty());
+        let violations = crate::run_for_slice(
+            valid_test.code.as_bytes(),
+            "tmp.rs",
+            ConfigBuilder::default()
+                .rule(&self.rule.meta.name)
+                .language(SupportedLanguage::Rust)
+                .build()
+                .unwrap(),
+        );
+        assert!(violations.is_empty());
     }
 
     fn run_invalid_test(&self, invalid_test: &RuleTestInvalid) {
-        let tmp_dir = TempDir::new("invalid_test").unwrap();
-        let test_filename = "tmp.rs";
-        fs::write(tmp_dir.path().join(test_filename), &invalid_test.code).unwrap();
-        Command::cargo_bin("tree-sitter-lint")
-            .unwrap()
-            .args(["--rule", &self.rule.name])
-            .current_dir(tmp_dir.path())
-            .assert()
-            .failure()
-            .code(1)
-            .stdout(predicate::function(|stdout: &str| {
-                does_invalid_output_match_expected(stdout, &invalid_test.errors)
-            }));
+        let mut file_contents = invalid_test.code.clone().into_bytes();
+        let violations = crate::run_fixing_for_slice(
+            &mut file_contents,
+            "tmp.rs",
+            ConfigBuilder::default()
+                .rule(&self.rule.meta.name)
+                .language(SupportedLanguage::Rust)
+                .fix(true)
+                .report_fixed_violations(true)
+                .build()
+                .unwrap(),
+        );
+        if let Some(expected_file_contents) = invalid_test.output.as_ref() {
+            assert_eq!(&file_contents, expected_file_contents.as_bytes());
+        }
```

Comment:
So per above comment if we want to enforce that "you always provide `output` for any test case that does fixing" that would go here